app/src/main/kotlin/io/homeassistant/companion/android/settings/server/ServerSettingsPresenterImpl.kt

```diff
@@ -192,4 +195,17 @@ class ServerSettingsPresenterImpl @Inject constructor(
     override suspend fun serverURL(): String? {
         return serverManager.getServer(serverId)?.connection?.getUrl()?.toString()
     }
+
+    override suspend fun securityLevelSummary(): Int? {
+        val allowInsecureConnection = serverManager.integrationRepository(serverId).getAllowInsecureConnection()
+        return when (allowInsecureConnection) {
```

Comment:
Returning the string from this class is a different pattern from most places, where the class returns the settings value and the fragment decides how to display it and set a summary (see: ServerSettingsView functions).